burn-wgpu/src/template/comparison/binary.wgsl

```diff
@@ -0,0 +1,37 @@
+@group(0)
+@binding(0)
+var<storage, read> lhs: array<{{ elem }}>;
+
+@group(0)
+@binding(1)
+var<storage, read> rhs: array<{{ elem }}>;
+
+@group(0)
+@binding(2)
+var<storage, read_write> output: array<u32>;
+
+@group(0)
+@binding(3)
+var<storage, read> info: array<u32>;
+
+
+@compute
+@workgroup_size({{ workgroup_size_x }}, 1, 1)
+fn main(@builtin(global_invocation_id) global_id: vec3<u32>) {
+    let dim: u32 = info[0];
+    var index_lhs: u32 = 0u;
+    var index_rhs: u32 = 0u;
+
+    for (var i: u32 = 0u; i < dim; i++) {
```

Comment:
If you replace by `for (var i: u32 = 1u; i <= dim+1; i++) {` you should be able to remove all the + 1u